app/src/androidTest/java/br/com/concrete/tentacle/base/BaseTest.kt

```diff
@@ -0,0 +1,98 @@
+package br.com.concrete.tentacle.base
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.test.espresso.IdlingRegistry
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import br.com.concrete.tentacle.util.LayoutChangeCallback
+import okhttp3.mockwebserver.MockWebServer
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.StringDescription
+import org.hamcrest.TypeSafeMatcher
+import org.junit.After
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.runner.RunWith
+
+@Ignore
+@RunWith(AndroidJUnit4::class)
+abstract class BaseTest {
+
+    lateinit var mockWebServer: MockWebServer
+    lateinit var testFragment: Fragment
+
+    @Before
+    fun setUp(){
+        mockWebServer = MockWebServer()
+        mockWebServer.start(8080)
+    }
+
+    @After
+    fun after() {
+        mockWebServer.shutdown()
+    }
+
+    fun getJson(path: String): String {
+        val json = InstrumentationRegistry.getInstrumentation()
+            .targetContext
+            .assets
+            .open(path).bufferedReader().use { it.readText() }
+
+        return json
+    }
+
+    fun waitUntil(matcher: Matcher<View>): ViewAction {
+        return ViewActions.actionWithAssertions(object : ViewAction {
+            override fun getConstraints(): Matcher<View> {
+                return ViewMatchers.isAssignableFrom(View::class.java)
+            }
+
+            override fun getDescription(): String {
+                val description = StringDescription()
+                matcher.describeTo(description)
+                return String.format("wait until: %s", description)
+            }
+
+            override fun perform(uiController: UiController, view: View) {
+                if (!matcher.matches(view)) {
+                    val callback = LayoutChangeCallback(matcher)
+                    try {
+                        IdlingRegistry.getInstance().register(callback)
+                        view.addOnLayoutChangeListener(callback)
+                        uiController.loopMainThreadUntilIdle()
+                    } finally {
+                        view.removeOnLayoutChangeListener(callback)
+                        IdlingRegistry.getInstance().unregister(callback)
+                    }
+                }
+            }
+        })
+    }
+
+    fun childAtPosition(
```

Comment:
acho que ficaria interessante criar uma extension function e adicionar esse metodo, ficar adicionando ele em uma Base por exemplo vai inchar muito a classe.

Reply:
se eu criar uma extension function vai tornar o código somente mais complexo, se você reparar na chamada dessa função, ela é aninhada para que eu possa encontrar um elemento dentro de outro, não acho que vale a pena fazer uma extension que dê 'suporte' a isso

Reply:
Na verdade o código iria continuar o mesmo, so o método que iria mudar de lugar, acho que ficaria até mais organizado so pela separação, to tipo, ter um arquivo onde vai ser RecyclerTestExtensionFunctions, onde vai ter por exemplo essa função de childAtPosition, onde voce ja vai bater o olho e saber que nesse arquivo tem os métodos customizados que eu posso utilizar para testar o recyclerview

Reply:
Isso pode ser uma coisa para vocês conversarem se isso começar a fazer sentido, para poder organizar o codigo de teste.

Reply:
Esse método não serve somente pra recyclerview, eu vou tentar mudar e ver se fica melhor mesmo 🗡